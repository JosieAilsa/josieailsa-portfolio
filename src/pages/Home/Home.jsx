import heroImg from "../../assets/images/landing-img.svg";
import "./Home.scss";
import scroll from "../../assets/images/icons/carousel-arrow2.svg";
import Button from "../../components/Button/Button";

const Home = () => {
  return (
    <div id="home" className="home page">
      <section className="hero">
        <section className="hero__text">
          <h1 className="hero__header">Hi, I'm Josie.</h1>
          <h2 className="hero__intro">
            I'm a Bristol-based developer.
          </h2>
          <Button
            classNames={"hero__button button--primary button"}
            content={<a href="#projects">See my projects</a>}
          />
        </section>
        <img className="hero__image" src={heroImg} alt="hero" />
      </section>
      <article className="scroll">
        <img className="scroll__image" src={scroll} alt="" />
        <p className="scroll__text">scroll</p>
      </article>
    </div>
  );
};

export default Home;
