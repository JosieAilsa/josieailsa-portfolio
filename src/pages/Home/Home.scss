@use "../../assets/sass/abstracts/variables" as *; 
@use "../../assets/sass/layout/layout" as *; 
@use "../../assets/sass/abstracts/mixin" as *;
@use"../../assets/sass/base/resets" as *;

.home{
    display: flex;
    flex-direction: column;
    justify-content: flex-end;
    align-items: center;
    @media only screen and (min-width: $desktop-breakpoint) {
    @include flex-center-column
    }
    .hero{
        @include flex-center-column;
        width: 100%;
        @media only screen and (min-width: $desktop-breakpoint) {
        display: flex;
        flex-direction: row-reverse;
        }
    
    .hero__image{
        width: 85%;
        padding: 2rem;
        @media only screen and (min-width: $tablet-breakpoint) {
            max-width: 65%;
            }
        @media only screen and (min-width: $desktop-breakpoint) {
            max-width: 35%;
            }
         
    }
    
    
    .hero__text{
        color: $color-green;
        @include flex-center-column;
        width: 70%;
        gap:1rem;
        text-align: center;
        @media only screen and (min-width: $desktop-breakpoint) {
            width: 50%;
            padding: 2rem;
        }
        .hero__header{
            font-family: $font-accent;
            font-size: 2.15rem;
            font-weight: 400;
            font-style: italic;
            @media only screen and (min-width: $desktop-breakpoint) {
            font-size: 5rem;
            text-align: left;
            white-space: nowrap;
            }
        }
        .hero__intro{
            font-family: $font-body;
            font-size: 1.25rem;
            font-weight: 500;

            @media only screen and (min-width: $desktop-breakpoint) {
                font-size: 2rem;
                text-align: center;
                }
            }
        }
        }
    }
  
