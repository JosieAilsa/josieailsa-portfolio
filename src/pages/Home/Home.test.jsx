import Home from "./Home";
import { customRender } from "../../utils/testutils";

it("The home page", () => {
  const { container } = customRender(<Home />);
  expect(container).toMatchSnapshot();
});
