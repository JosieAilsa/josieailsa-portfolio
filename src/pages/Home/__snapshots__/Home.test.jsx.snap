// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The home page 1`] = `
<div>
  <div
    class="home page"
  >
    <section
      class="hero"
    >
      <section
        class="hero__text"
      >
        <h1
          class="hero__header"
        >
          Hi, I'm Josie.
        </h1>
        <h3
          class="hero__intro"
        >
          I'm Bristol-based UI developer and software coach.
        </h3>
        <button
          class="hero__button hero__button--active"
          role="button"
        >
          See my projects
        </button>
      </section>
      <img
        alt="hero image"
        class="hero__image"
        src="landing-img.svg"
      />
    </section>
  </div>
</div>
`;
